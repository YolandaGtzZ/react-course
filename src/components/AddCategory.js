import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setinputValue] = useState("");

  const handleInputChange = (e) => {
    setinputValue(e.target.value);

    console.log("HandleInputChange llamado");
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('handleSubmit');

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setinputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <div>
        <input type="text" value={inputValue} onChange={handleInputChange} />
      </div>
    </form>
  );
};

AddCategory.protoTypes = {
  setCategories: PropTypes.func.isRequired,
};
