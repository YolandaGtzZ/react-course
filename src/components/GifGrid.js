import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  useFetchGifs.mockReturnValue({
    data:[ ],
    loading: true
  });

  const { data:images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__swing'>{category}</h3>
      {loading && <p>Loading</p>}
      <div >
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};


GifGrid.propTypes ={
  category: PropTypes.string.isRequired
};

