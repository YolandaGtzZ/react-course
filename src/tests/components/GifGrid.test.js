import React from 'react';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import {shallow} from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba del componente <GifGrid/>', () => {

    const category = 'One Punch';

    test('Tomar el snapshot del componente', () => {
      const wrapper = shallow(<GifGrid category ={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan las imagenes', () => {
       const gifs = [{
           id: 'ABC',
           url: 'https://localhost.com',
           title:'Cualquier cosa',
       }];
       
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: true
          });
        
        const wrapper = shallow(<GifGrid category ={category}/>);
        expect(wrapper).toMatchSnapshot();
       expect(wrapper.find('p').exists()).toBe(false);


    })
    
    
})
 