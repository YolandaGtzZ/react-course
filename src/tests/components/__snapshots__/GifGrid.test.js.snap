// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <GifGrid/> Tomar el snapshot del componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__swing"
  >
    One Punch
  </h3>
  <p>
    Loading
  </p>
  <div />
</Fragment>
`;

exports[`Prueba del componente <GifGrid/> debe de mostrar items cuando se cargan las imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__swing"
  >
    One Punch
  </h3>
  <p>
    Loading
  </p>
  <div />
</Fragment>
`;
